### PR TITLE
Fixed SNI for Python 2.7.9+

### PR DESCRIPTION
**Pull request by ehsanf**

### CODE REVIEW

**Comment by emintham** [comment]
Why are we duplicating yantra code in sdetools? Is there a problem with using it as a dependency?

**Comment by geoffwhittington** [comment]
@emintham sdetools stopped being portable when yantra became an external dependency (check the extlib directory, we have put a bunch of dependencies in there).

**Comment by funkaoshi** [comment]
I also think we should get away from "vendoring" all our dependencies. Is the issue that yantra isn't up on pypi? That's one problem right now. You can't just pip install sdetools.

File: sdetools/sdelib/restclient.py [hunk 1]
```diff
@@ -299,12 +299,22 @@ def call_api(self, target, method=URLRequest.GET, args=None, call_headers={}, au
...
+                    type(err.reason) is http_req.ssl.SSLError):
+                raise ServerError('Unable to verify SSL certificate for host: %s (reason: %s)' % 
+                        (self.server, err.reason))
             handle = err
             call_success = False
         except httplib.InvalidURL, err:
             raise UsageError('Invalid URL. Reason %s' % (err))
+        except Exception as err:
+            if http_req.ssl_lib_found and type(err) is http_req.ssl.CertificateError:
+                raise ServerError('Unable to verify SSL certificate for host: %s (reason: %s)' %
```
Comment (funkaoshi) [inline sdetools/sdelib/restclient.py]: Can make a new exception SSLError or something that takes self.server & err.reason as params so you don't have to repeat your message all over the place.

**Comment by funkaoshi** [comment]
@emintham you may need to wrap this up. I think @ehsanf pushing this up at all was a one time thing. Ha

**Comment by geoffwhittington** [comment]
The original practice of putting all its dependencies into the extlib directory allowed us to: 1) make sure it had everything it needed to run and 2) fork & deploy customized 3rd party libraries (SOAPpy for one).

**Pull request merged**